stk/molecular/molecules/molecule.py

```diff
@@ -566,55 +570,62 @@ def mdl_mol_block(self, conformer=-1):
         except ValueError:
             pass
 
-        n_atoms = self.mol.GetNumAtoms()
-        n_bonds = self.mol.GetNumBonds()
-
-        dtype = np.dtype(object)
-
-        atom_ids = np.array([np.arange(1, n_atoms+1)],
-                            dtype=dtype).T
-
-        atom_symbols = np.array([[self.atom_symbol(i)]
-                                 for i in range(n_atoms)],
-                                dtype=dtype)
-
-        pos_mat = self.mol.GetConformer(conformer).GetPositions()
-
-        charges = np.array([[f' CHG={a.GetFormalCharge()}' if
-                             a.GetFormalCharge() else '']
-                            for a in self.mol.GetAtoms()],
-                           dtype=dtype)
-
-        atom_data = np.concatenate(
-                        [atom_ids, atom_symbols, pos_mat, charges],
-                        axis=1).reshape((-1, ))
-        atom_block = "M  V30 {} {} {:.4f} {:.4f} {:.4f} 0{}\n"*n_atoms
-        atom_block = atom_block.format(*atom_data)
-
-        bond_data = [prop for bond in self.mol.GetBonds() for prop in
-                     (bond.GetIdx(),
-                      int(bond.GetBondTypeAsDouble()),
-                      bond.GetBeginAtomIdx()+1,
-                      bond.GetEndAtomIdx()+1)]
-        bond_block = "M  V30 {} {} {} {}\n"*n_bonds
-        bond_block = bond_block.format(*bond_data)
-
-        return ("\n"
-                "     RDKit          3D\n"
-                "\n"
-                "  0  0  0  0  0  0  0  0  0  0999 V3000\n"
-                "M  V30 BEGIN CTAB\n"
-                f"M  V30 COUNTS {n_atoms} {n_bonds} 0 0 0\n"
-                "M  V30 BEGIN ATOM\n"
-                f"{atom_block}"
-                "M  V30 END ATOM\n"
-                "M  V30 BEGIN BOND\n"
-                f"{bond_block}"
-                "M  V30 END BOND\n"
-                "M  V30 END CTAB\n"
-                "M  END\n"
-                "\n"
-                "$$$$\n")
+        if atoms is None:
+            atoms = range(self.mol.GetNumAtoms())
```

Comment:
I think it might fail if one starts adding and removing atoms as there might be some missing id's (unless rdkit takes care of it differently than with conformers). Might be safer to get a sequence of all atoms by `self.mol.GetAtoms()`.

Reply:
It is different. This should be safe, with atoms you identify them by their index (.GetAtomWithIdx()).